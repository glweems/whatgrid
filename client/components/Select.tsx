import * as React from 'react';
import uuid from 'uuid/v4';
import { Select } from '@rebass/forms';

interface Props extends React.SelectHTMLAttributes<HTMLSelectElement> {
  options: string[];
}

export default ({ options, ...props }: Props) => {
  return (
    <Select {...props}>
      {options.map(o => (
        <option key={uuid()} value={o}>
          {o}
        </option>
      ))}
    </Select>
  );
};
