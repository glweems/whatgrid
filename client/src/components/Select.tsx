import React from 'react';
import uuid from 'uuid/v4';

interface Props extends React.SelectHTMLAttributes<HTMLSelectElement> {
  options: string[];
}

const Select = ({ options, ...props }: Props): React.ReactElement<HTMLSelectElement> => {
  return (
    <select {...props}>
      {options.map((o) => (
        <option key={uuid()} value={o}>
          {o}
        </option>
      ))}
    </select>
  );
};

export default Select;
