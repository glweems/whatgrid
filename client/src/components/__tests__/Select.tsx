import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Select from '../Select';

describe('Select', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Select options={['one', 'two', 'three']} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
