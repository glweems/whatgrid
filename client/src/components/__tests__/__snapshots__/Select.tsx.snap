// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<select>
  <option
    value="one"
  >
    one
  </option>
  <option
    value="two"
  >
    two
  </option>
  <option
    value="three"
  >
    three
  </option>
</select>
`;
